### PR TITLE
Build-Your-Own Workshop Enrollment updates

### PR DESCRIPTION
**Pull request by TurnerRiley**
This PR makes some updates to the Build Your Own workshop enroll form. Previously, when a user navigated to the enroll form url for a Build Your Own workshop, they were auto-enrolled and sent straight to the My PL page. As part of the new Build Your Own workshop work, we want to:
- Show the enroll form but with no questions, just workshop details and the "Register" button
- Show the Build Your Own workshop's self-paced PL topics in place of where "Course" is shown in the details
- Make the "Register" button purple (for all enroll forms)

### BYOW enroll form
byow updates

### Still shows success message upon enrollment
success message

## Links
Jira ticket to only show the workshop information and register button for BYO enroll forms: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2630)
Jira ticket to list PL topics instead of courses in workshop enroll form description: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2634)
Jira ticket to update 'Register' button to be purple: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2633)
Jira ticket to ensure BYO workshops show confirmation messages on enrollment: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2631)

## Testing story
Local testing and adding a unit test.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx
Comment (TurnerRiley) [inline apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx]: Turn whitespace off to see the changes more easily!

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx
Comment (bethanyaconnor) [inline apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: Shouldn't the default have white text? Do you know why that isn't working correctly?

Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: That's what I thought too but it was showing black, this file doesn't import styles though so that's why I just hard-coded it here since other components were hard-coding styles.

Comment (drizco) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: I'm in the middle of refactoring this form to use dsco components, and I found bootstrap imported in its stylesheet. I'm guessing that's the culprit. removing it in my branch, so once this code merges I can clean it up

Screenshot 2024-11-18 at 2 01 13 PM

Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: Okay sounds good thank you!

Comment (bethanyaconnor) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: Cool, so Ryan will clean this up? That SGTM!

I know this is a nit, but I'd rather not keep unnecessary inline styling around!

**Review by bethanyaconnor — APPROVED** [review]

**Comment by bethanyaconnor** [comment]
Out of curiousity, does this fix the issue described in https://codedotorg.atlassian.net/browse/ACQ-2631? (Approval still good if it doesn't, just want to understand!)

**Comment by TurnerRiley** [comment]
> Out of curiousity, does this fix the issue described in https://codedotorg.atlassian.net/browse/ACQ-2631? (Approval still good if it doesn't, just want to understand!)

Yes this still covers it! Since the BYO workshops auto-enrolled, they would just go straight to the confirmation message but I believe this ticket is ensuring that with the new enroll form now showing that the confirmation still shows (which it does)!

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx
Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: I could wait for Ryan's fix then merge it in without the inline change? Otherwise I think we'll have to decide between some temporarily inline styling or temporarily having the button text be black?

Comment (drizco) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: <s>oh, another option might be to swap in the dsco buttons right away?</s>

oops, already doing that!

my pr is turning out to be a pretty drastic refactor, so I think it would be easier if I handle the conflicts 😬

Comment (bethanyaconnor) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: fwiw, temporary inline styling is completely fine to me. Do whatever is easier for you both :)

Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx:885]: Okay sounds good I'll merge with the temporarily inline styling but let Ryan move forward with it to avoid more conflicts!

**Pull request merged**
